### PR TITLE
Feature placeholder

### PR DESCRIPTION
**Pull request by LucaBecea**
Add a customizable placeholder prop to react combo that can be a string or a function as follows:
```
<Combo
  placeholder="this is a placeholder"
>
```
or
```
<Combo
  placeholder={this.renderPlaceholder} 
>

renderPlaceholder(){
  return <span>this is a placeholder</span>
}
```

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -202,6 +208,15 @@ export default class Combo extends Component {
     </div>
   }
 
+  renderPlaceholder() {
+    const placeholder = this.props.placeholder
+    if (typeof placeholder == 'string') {
```
Comment (bogdanpetru) [inline src/index.js]: @LucaBecea placeholder can also be jsx

Comment (LucaBecea) [inline src/index.js]: Done

File: src/index.js [hunk 2]
```diff
@@ -211,10 +211,17 @@ export default class Combo extends Component {
...
       return <div className="react-combo__placeholder" >{placeholder}</div>
     } else if (typeof placeholder == 'function'){
-      return placeholder()
+      const placeholderElement = React.createElement(placeholder)
+
+      if (placeholderElement.type.length) {
+        return placeholderElement
+      } else {
+        return placeholder()
+      }
```
Comment (bogdanpetru) [inline src/index.js]: @LucaBecea 
It is simpler to check only it if is a function. Also what comes out from the function should go in the  ```<div className="react-combo__placeholder" >```
```
if (typeof placeholder === 'function) {
  return  <div className="react-combo__placeholder" >{placeholder()}</div>
} else {
   return <div className="react-combo__placeholder" >{placeholder}</div>
}

Comment (radubrehar) [inline src/index.js]: Do not duplicate the code.

```jsx
const children = typeof placeholder == 'function'? placeholder(): placeholder

return <div className="react-combo__placeholder" children={children} />
```

Comment (radubrehar) [inline src/index.js]: @LucaBecea please update accordingly

File: src/index.js [hunk 3]
```diff
@@ -166,9 +166,11 @@ export default class Combo extends Component {
   render(){
     const props = this.p = this.prepareProps(this.props)
     const expanded = props.expanded
+    const placeholderActive = props.selectedItems.length == 0 && this.state.text == '' && props.placeholder? true : false
```
Comment (radubrehar) [inline src/index.js]: @LucaBecea please implement a `isPlaceholderVisible` method.

File: src/index.js [hunk 4]
```diff
@@ -166,9 +166,11 @@ export default class Combo extends Component {
...
 
     const list = this.renderList(props)
-    const tags = this.renderTags(props)
+    const placeholder = this.renderPlaceholder()
```
Comment (radubrehar) [inline src/index.js]: The `renderPlaceholder` function should check in its body of the placeholder is active or not. If it's not, it should return null.

File: src/index.js [hunk 5]
```diff
@@ -183,13 +185,18 @@ export default class Combo extends Component {
...
       {...props}
+      placeholder={null}
       data={null}
       tabIndex={tabIndex}
       onFocus={this.onFocus}
       onBlur={null}
       onMouseDown={this.onMouseDown}
     >
       {tags}
+      {
```
Comment (radubrehar) [inline src/index.js]: @LucaBecea after the above changes, just render `{placeholder}`

**Pull request merged**